articles/steganography-in-python/index.md

```diff
@@ -0,0 +1,248 @@
+author: lalithnarayan-c
+date: 2020-08-18T00:00:00-07:00
+topics: []
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/steganography-in-python/hero.jpg
+    alt:
+---
+In steganography, an innocent-looking image is taken as an example and a message is embedded in the image by changing the number of pixels selected by the encryption algorithm. By embedding a hidden message or file in an image, the number of pixels can be changed.
+
+In short, this means we use the encrypted RGB data to include other data, which significantly impairs the visual representation of the image. The hidden message is transmitted by increasing the bandwidth of the original message or by manipulating the file format. Since steganography is often used in phishing and as a way for malicious software to exfiltrate data, it is very difficult to detect it.
+<!--more-->
+
+### Military Uses vs Malicious Threats
+To understand how image steganography works, we need to take a look at some of the basic ways in which text can be hidden in image files. Due to the nature of image file formats, it is not only possible to hide text strings, but also to hide entire files in jpg or other image formats. In a technique known as secret-key steganography, steganography is applied to a message that already has a key. This ensures multiple levels of security.
+
+In malware, image steganography can be used to hide payloads within the code itself, or it can call additional code or executable programs associated with the attack. The attacker can also identify and replace the methods of the steganography in other ways, for example in the form of a command and control (C & C) script. The recent attack on Jeff Bezos's phone was one such act. The code was maliciously placed under the media file. Upon downloading, the code started executing in the background and transferred all the data illegally.
+
+There are several [simple detection methods](https://0xrick.github.io/lists/stego/) based on file scans, such as the steganography tools found on the Internet.  But they are poorly developed and not extensive enough to detect such malicious acts.
+
+Military intelligence services, however, use high-quality proprietary tools allowing automation to the process of detecting steganography. A lot of analytical tools can also be used if there is a suspicion that stealth is being used to hide messages.
+
+### Cryptography Vs Steganography
+
+As a technique for hiding data, cryptography is also used for the secure transmission of messages over the Internet. Cryptography is the science of ensuring that only the receiver (of the intended message) reads the message. Steganography, on the other hand, is the science of intelligently placing information so that no one knows about its presence. There are several cases where steganography may be better suited than cryptography.
```

Comment:
Perhaps add that they're not mutually exclusive. E.g. use cryptography to protect a message hidden by steganography so that if its presence is discovered, it is still unreadable.

Reply:
Thank you for your feedback. Have included the changed in the latest commit.